src/Queries/GeoshapeQuery.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Spatie\ElasticsearchQueryBuilder\Queries;
+
+use Webmozart\Assert\Assert;
+
+class GeoshapeQuery implements Query
+{
+    public const TYPE_POINT = 'point';
+    public const TYPE_LINESTRING = 'linestring';
+    public const TYPE_POLYGON = 'polygon';
+    public const TYPE_MULTIPOINT = 'multipoint';
+    public const TYPE_MULTILINESTRING = 'multilinestring';
+    public const TYPE_MULTIPOLYGON = 'multipolygon';
+    public const TYPE_GEOMETRYCOLLECTION = 'geometrycollection';
+    public const TYPE_ENVELOPE = 'envelope';
+    public const RELATION_INTERSECTS = 'intersects';
+    public const RELATION_DISJOINT = 'disjoint';
+    public const RELATION_CONTAINS = 'contains';
+    public const RELATION_WITHIN = 'within';
+
+    protected string $field;
+    protected string $type;
+    protected array $coordinates;
+    protected ?string $relation;
+
+    public function __construct(
+        string $field,
+        string $type,
+        array $coordinates,
+        ?string $relation = self::RELATION_INTERSECTS,
+    ) {
+        $this->field = $field;
+        $this->coordinates = $coordinates;
+
+        Assert::oneOf($type, [
```

Comment:
Sorry for the back and forth, but I don't believe webmozart's assert library is part of our production dependencies. Could we either remove this or solve it without a dev dependency?

Reply:
No worries, it's my bad. I've updated this now.